packages/swap/src/utils/statechain.ts

```diff
@@ -11,17 +11,27 @@ export type SwapRateArgs = {
   srcAsset: InternalAsset;
   destAsset: InternalAsset;
   amount: bigint;
+  limitOrders?: LimitOrders;
 };
 
-export const getSwapRate = async ({ srcAsset, destAsset, amount }: SwapRateArgs) => {
+export type LimitOrders = RpcParams['cf_swap_rate_v2'][3];
+
+export const getSwapRateV2 = async ({ srcAsset, destAsset, amount, limitOrders }: SwapRateArgs) => {
   const client = initializeClient();
 
-  const { intermediary, output } = await client.sendRequest(
-    'cf_swap_rate',
+  const {
+    intermediary: intermediateAmount,
+    output: outputAmount,
+    egress_fee: egressFee,
+    ingress_fee: ingressFee,
+    network_fee: networkFee,
+  } = await client.sendRequest(
+    'cf_swap_rate_v2',
     getAssetAndChain(srcAsset),
     getAssetAndChain(destAsset),
     hexEncodeNumber(amount),
+    limitOrders,
```

Comment:
It would be nice to validate the response against a zod schema

Reply:
This is the new RPC package's RPC client, it validates the response already for us